### PR TITLE
Update i18n code to use hooks

### PR DESCRIPTION
**Pull request by leovalette**
This PR allows using the [useTranslation hook](https://react.i18next.com/latest/usetranslation-hook)
  
  (Sorry, there is a lot of "noises" due to our different VSCode settings)

### CODE REVIEW

**Review by Nas84 — APPROVED** [review]
LGTM

File: packages/yourloops/app/app.tsx [hunk 1]
```diff
@@ -26,72 +26,51 @@
...
-      document.body.appendChild(div);
-    }
-    renderDOM(router, div);
-  }
+import { AuthProvider } from "../lib/auth/hook/use-auth";
+import HcpPage from "../pages/hcp";
+import LoginPage from "../pages/login";
+import PatientPage from "../pages/patient";
+import PrivateRoute from "../components/private-route";
+import { Suspense } from "react";
```
Comment (yvon-dblg) [inline packages/yourloops/app/app.tsx:46]: Avoid 2 react imports

File: packages/yourloops/app/app.tsx [hunk 2]
```diff
@@ -26,72 +26,51 @@
...
               <PrivateRoute path="/hcp" component={HcpPage} />
               <PrivateRoute path="/patient" component={PatientPage} />
             </Switch>
           </Router>
         </AuthProvider>
       </ThemeProvider>
-    );
-  }
-}
+    </Suspense>
```
Comment (yvon-dblg) [inline packages/yourloops/app/app.tsx:72]:
```suggestion
    </React.Suspense>
```

File: packages/yourloops/app/app.tsx [hunk 3]
```diff
@@ -26,72 +26,51 @@
...
+import PrivateRoute from "../components/private-route";
+import { Suspense } from "react";
+import { ThemeProvider } from "@material-ui/core/styles";
+import { theme } from "../components/theme";
 
-  private router(): JSX.Element {
-    return (
+const Yourloops: React.FunctionComponent = () => {
+  return (
+    <Suspense fallback="loading">
```
Comment (yvon-dblg) [inline packages/yourloops/app/app.tsx:52]:
```suggestion
    <React.Suspense fallback="loading">
```

File: packages/yourloops/app/app.tsx [hunk 4]
```diff
@@ -26,72 +26,51 @@
...
+import LoginPage from "../pages/login";
+import PatientPage from "../pages/patient";
+import PrivateRoute from "../components/private-route";
+import { Suspense } from "react";
+import { ThemeProvider } from "@material-ui/core/styles";
+import { theme } from "../components/theme";
 
-  private router(): JSX.Element {
-    return (
+const Yourloops: React.FunctionComponent = () => {
```
Comment (yvon-dblg) [inline packages/yourloops/app/app.tsx:50]: Use a plain function, no need to declare a var as a function here

File: packages/yourloops/lib/language.ts [hunk 5]
```diff
@@ -101,10 +102,7 @@ i18n.on('languageChanged', (lng: string) => {
   }
 });
 
-async function initI18n(): Promise<TFunction> {
-  return i18n.init(i18nOptions);
-}
-
+i18n.init(i18nOptions);
```
Comment (yvon-dblg) [inline packages/yourloops/lib/language.ts:105]: The init function is a promise, may be async, should be taken into account

File: packages/yourloops/pages/hcp/patients-list-bar.tsx [hunk 6]
```diff
@@ -27,14 +27,20 @@
  */
 
 import * as React from "react";
-import { useHistory } from "react-router-dom";
 
+import { FilterType, Team } from "./types";
+import { REGEX_EMAIL, defer } from "../../lib/utils";
```
Comment (yvon-dblg) [inline packages/yourloops/pages/hcp/patients-list-bar.tsx:32]: Local imports at the last position is the best !

File: packages/yourloops/pages/password-reset/request-password-reset.tsx [hunk 7]
```diff
@@ -37,26 +35,31 @@ import CardContent from "@material-ui/core/CardContent";
...
+import { RouteComponentProps } from "react-router-dom";
 import TextField from "@material-ui/core/TextField";
 import Typography from "@material-ui/core/Typography";
-
-import { t } from "../../lib/language";
+import _ from "lodash";
 import brandingLogo from "branding/logo.png";
-import { useState } from "react";
 import { useAuth } from "../../lib/auth/hook/use-auth";
+import { useState } from "react";
```
Comment (yvon-dblg) [inline packages/yourloops/pages/password-reset/request-password-reset.tsx:44]:
```suggestion
```

**Review by yvon-dblg — CHANGES_REQUESTED** [review]
General remarks: 
- Too much code "noise", not needed on this PR.
- Local import have to be on last positions, even if the auto tools do it otherwise.
- Avoid 2 imports of the same lib (e.g. "react")

File: packages/yourloops/app/app.tsx (hunk 4 above)
Comment (leovalette) [reply packages/yourloops/app/app.tsx:50]: Why do you prefer plain function over arrow function ?

File: packages/yourloops/lib/language.ts (hunk 5 above)
Comment (leovalette) [reply packages/yourloops/lib/language.ts:105]: This is why we need the Suspense wrapper around the app

File: packages/yourloops/pages/hcp/patients-list-bar.tsx (hunk 6 above)
Comment (leovalette) [reply packages/yourloops/pages/hcp/patients-list-bar.tsx:32]: Totally agreed, I forgot to setup my imports organizer

File: packages/yourloops/pages/password-reset/request-password-reset.tsx (hunk 7 above)
Comment (leovalette) [reply packages/yourloops/pages/password-reset/request-password-reset.tsx:44]: That's why I don't like `import * as React from "react";` This makes the code much heavier.

File: packages/yourloops/lib/language.ts (hunk 5 above)
Comment (yvon-dblg) [reply packages/yourloops/lib/language.ts:105]: This page describes experimental features that are not yet available in a stable release.
https://reactjs.org/docs/concurrent-mode-suspense.html

File: packages/yourloops/pages/password-reset/request-password-reset.tsx (hunk 7 above)
Comment (yvon-dblg) [reply packages/yourloops/pages/password-reset/request-password-reset.tsx:44]: For me writing `React.useState()` is cleaner anyway 😉

File: packages/yourloops/app/app.tsx (hunk 4 above)
Comment (yvon-dblg) [reply packages/yourloops/app/app.tsx:50]: I think for "globale" module functions, the code the more readable.

File: packages/yourloops/pages/password-reset/request-password-reset.tsx (hunk 7 above)
Comment (leovalette) [reply packages/yourloops/pages/password-reset/request-password-reset.tsx:44]: I don't get why, but as you wish 😄

File: packages/yourloops/lib/language.ts (hunk 5 above)
Comment (leovalette) [reply packages/yourloops/lib/language.ts:105]: This is the solution given by [react.i18n](https://react.i18next.com/latest/using-with-hooks). Should we find another solution ?

**Review by yvon-dblg — APPROVED** [review]
LGTM

**Pull request merged**